test/test_rosbag/bag_migration_tests/CMakeLists.txt

```diff
@@ -5,22 +5,82 @@
 # generate_messages() it might make it work, or we can migrate the bag
 # files themselves.
 
-add_message_files(
-  DIRECTORY msg_current
-  FILES
-  Constants.msg
-  Converged.msg
-  MigratedAddSub.msg
-  MigratedExplicit.msg
-  MigratedImplicit.msg
-  MigratedMixed.msg
-  PartiallyMigrated.msg
-  Renamed4.msg
-  Simple.msg
-  SimpleMigrated.msg
-  SubUnmigrated.msg
-  Unmigrated.msg
-)
+set(MSG_DIRECTORY "msg_current" CACHE STRING "Messages directory")
+message(STATUS "Building messages in: ${MSG_DIRECTORY}")
+
+if(${MSG_DIRECTORY} STREQUAL "msg_gen1")
+  add_message_files(
+    DIRECTORY ${MSG_DIRECTORY}
+    FILES
+    Constants.msg
+    Converged.msg
+    Convergent.msg
+    MigratedAddSub.msg
+    MigratedExplicit.msg
+    MigratedImplicit.msg
+    MigratedMixed.msg
+    PartiallyMigrated.msg
+    Renamed1.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
```

Comment:
Done.

I create a list of common messages, and separate list for each generation. Then I use the `MSG_DIRECTORY` macro to select the list directly, with no if-else statements. Now is much more compact and cleaner.